include/highfive/bits/H5FileDriver_misc.hpp

```diff
@@ -15,6 +15,7 @@ namespace HighFive {
 #ifdef H5_HAVE_PARALLEL
 inline MPIOFileDriver::MPIOFileDriver(MPI_Comm comm, MPI_Info info) {
     add(MPIOFileAccess(comm, info));
+    add(MPIOCollectiveMetadataWrite(true));
```

Comment:
Looks good. Only question I have: do we want this by default? Given that changes to the file structure have to be collective, it seems sensible. Just wondering if there are other implications.

Reply:
Fair enough, we don't want to support that driver anymore. Therefore, changing it in any way can only break things.

Reply:
I agree that collective metadata reads and writes should be opt-in, just as they are in HDF5 itself.   In my domain, I'm always behaving in a collective-friendly manner but I know that is not always the case with HDF5 users.